docs/topics/exceptions.rst

```diff
@@ -77,3 +78,25 @@ NotSupported
 
 This exception is raised to indicate an unsupported feature.
 
+StopDownload
+-------------
+
+.. versionadded:: 2.2
+
+.. exception:: StopDownload(fail=False)
+
+Raised from a :class:`~scrapy.signals.bytes_received` signal handler to
+indicate that no further bytes should be downloaded for a response.
+
+The ``fail`` boolean parameter indicates whether or not the resulting partial
+response should be handled by the request errback. In this case, the response
+will be available in the ``response`` attribute of the received
+:class:`~twisted.python.failure.Failure` object.
+``fail`` is ``False`` by default, which means the response will be handled
+by the request callback. Note that ``fail`` is a keyword-only argument.
+
+Regardless of which method handles the response, it will contain
+``"download_stopped"`` in its :attr:`Response.flags` attribute.
```

Comment:
```suggestion
The ``fail`` boolean parameter indicates whether or not the resulting partial
response should be handled by the request callback (fail=False, default) or by
the request errback (fail=True).

* If ``StopDownload(fail=False)`` is raised, request callback is called. Response
  which is passed to the callback could have its body truncated: the body
  contains only bytes received before the StopDownload exception is raised.
* When ``StopDownload(fail=True)`` exception is raised, request errback is called. 
  In this case response (possibly with a partial body) is available as the 
  ``response`` attribute of the received :class:`~twisted.python.failure.Failure` 
  object.

Note that ``fail`` is a keyword-only argument.

Regardless of which method handles the response, the response object has
``"download_stopped"`` in its :attr:`Response.flags` attribute.
```

Reply:
While writing the suggestion above, I realized we don't specify which data would response.body actually contain. In particular, would it include the current data chunk received in bytes_received signal?

Reply:
Updated, thanks!